site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -217,28 +395,72 @@ console.log(navigation.currentEntry.getState());
 ```
 
 By default, this will be `undefined`.
-You can synchronously set the state for the current `NavigationHistoryEntry` by calling:
+
+#### Setting state
+
+Although state objects can be mutated, those changes are not saved back with the history entry, so:
 
 ```js
-navigation.updateCurrentEntry({state: something});
+const state = navigation.currentEntry.getState();
+console.log(state.count); // 1
+state.count++;
+console.log(state.count); // 2
+// But:
+console.info(navigation.currentEntry.getState().count); // will still be 1
 ```
 
-You can also set the state when navigating programmatically with `navigation.navigate()` (this is [described below](#programmatic-navigation)).
+The correct way to set state is during script navigation:
+
+```js
+navigation.navigate(url, {state: newState});
+// Or:
+navigation.reload({state: newState});
+```
 
-In the Navigation API, the state returned from `.getState()` is a copy of the previously set state.
-If you modify it, the stored version won't also change.
-For example:
+Where `newState` can be any [clonable object](https://developer.mozilla.org/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types).
+
+If you want to update the state of the current entry, it's best to perform a navigation that replaces the current entry:
 
 ```js
-navigation.updateCurrentEntry({state: {count: 1}});
+navigation.navigate(location.href, {state: newState, history: 'replace'});
+```
 
-const state = navigation.currentEntry.getState();
-state.count = 2;
+Then, your `"navigate"` event listener can pick up this change via `navigateEvent.destination`:
 
-console.info(navigation.currentEntry.getState().count); // will still be one
+```js
+navigation.addEventListener('navigate', navigateEvent => {
+  console.log(navigateEvent.destination.getState());
+});
 ```
 
-### Access All Entries
+{% Aside %}
+It's also possible to synchronously update state without triggering a navigation, using `updateCurrentEntry()`:
+
+```js
+navigation.updateCurrentEntry({state: newState});
+```
+
+Then, other code can hear about this change via the `"currententrychange"` event:
+
+```js
+navigation.addEventListener('currententrychange', () => {
+  console.log(navigation.currentEntry.getState());
+});
+```
+
+However, this splits your state-handing code between the `"navigate"` event and the `"currententrychange"` event.
+It's generally better to handle state in one place, the `"navigate"` event.
```

Comment:
I appreciate the deemphasis on updateCurrentEntry(). However I hesitate a bit to leave this section without mentioning the react-to vs. initiate distinction that https://github.com/wicg/navigation-api#setting-the-current-entrys-state-without-navigating mentions. I'm not sure how to do that without expanding the section and thus perhaps overemphasizing updateCurrentEntry()... maybe just an explainer link?

Reply:
I'll work it into a new section but with caveats